services/compute/virtualmachine/virtualmachine.go

```diff
@@ -209,11 +210,15 @@ func (c *client) getWssdVirtualMachineHardwareConfiguration(vm *compute.VirtualM
 					assignment = wssdcommon.AssignmentType_GpuPV
 				case compute.GpuDefault:
 					assignment = wssdcommon.AssignmentType_GpuDefault
+				default:
+					return nil, errors.Wrapf(errors.InvalidInput, "Invalid GPU Assignment %s", *gpu.Assignment)
 				}
-				vmGPU := &wssdcommon.VirtualMachineGPU{
-					Assignment:      assignment,
-					PartitionSizeMB: *gpu.PartitionSizeMB,
-					Name:            *gpu.Name,
+				vmGPU.Assignment = assignment
+				if gpu.Name != nil {
+					vmGPU.Name = *gpu.Name
+				}
+				if gpu.PartitionSizeMB != nil {
```

Comment:
Is PartitionSizeMB a optional parameter?

Reply:
It's not an optional parameter. Should I remove this check?

Reply:
If it is not optional , should we not be checking if both the parameters are set?

Reply:
Yes, we should fail if assignment type is GPU-P and partition size is not specified.

Reply:
Apologies for the confusion. For Arc VMs, we need to provide partition size from portal and CLI because that's how we calculate the number of partitions or number of DDA devices. But for other platforms, it can be empty too depending on how they define and implement in their operators/CLI.
 
But from moc perspective, since we are getting vmGPUs array in hardware profile, there is no need for partition size to be mandatory.
 
So, from moc perspective, name and partition size parameters are optional. AssignmentType is required.

Reply:
So what would it mean if we dont set the partition size?

Reply:
For GPU-P homogeneous env, there is only one partition size available, so that will be used.

For GPU-P heterogeneous env, we will return an error. 

For GPU-DDA, partition size is not required/ not applicable.